### PR TITLE
fix memory errors

### PR DESCRIPTION
**Pull request by marler8997**
Fix a few memory errors I found while working on the new allocator interface.

### CODE REVIEW

File: lib/std/math/big/int.zig [hunk 1]
```diff
@@ -670,7 +673,8 @@ pub const Mutable = struct {
                 try t_big.add(r.toConst(), t_big.toConst());
 
                 // u = Cx + Dy, r as u
-                try x.mul(x.toConst(), Cp);
```
Comment (marler8997) [inline lib/std/math/big/int.zig:673]: The issue here is that during the `mul` operation, the current `limbs` array for `x` can be reallocated, which can invalidate the array that is stored in the resulting expression from `x.toConst()`.  My new allocator was setting memory that has been freed to `uninitialized` which exposed this bug.

File: test/tests.zig [hunk 2]
```diff
@@ -648,8 +648,9 @@ pub const StackTracesContext = struct {
 
             const stdout = child.stdout.?.inStream().readAllAlloc(b.allocator, max_stdout_size) catch unreachable;
             defer b.allocator.free(stdout);
-            var stderr = child.stderr.?.inStream().readAllAlloc(b.allocator, max_stdout_size) catch unreachable;
-            defer b.allocator.free(stderr);
+            var stderrFull = child.stderr.?.inStream().readAllAlloc(b.allocator, max_stdout_size) catch unreachable;
```
Comment (marler8997) [inline test/tests.zig:651]: This modifies the code to pass the same buffer to `free` that was allocated.

File: lib/std/fs/path.zig [hunk 3]
```diff
@@ -1110,7 +1110,7 @@ pub fn relativePosix(allocator: *Allocator, from: []const u8, to: []const u8) ![
         }
         if (to_rest.len == 0) {
             // shave off the trailing slash
-            return result[0 .. result_index - 1];
+            return allocator.shrink(result, result_index - 1);
```
Comment (marler8997) [inline lib/std/fs/path.zig:1113]: Code needs to shrink the buffer so that the length is correct when free is called on the buffer later.

Comment (andrewrk) [reply lib/std/fs/path.zig:1113]: This doesn't need to block the merge, but this would be an excellent source code comment rather than GitHub comment

**Comment by andrewrk** [comment]
Great work, thank you!

**Pull request merged**